File: client/Weapon.js
//
// Weapon.js
//

function Weapon(tag, maxclip, maxres)
{
	this.tag = tag;

	this.maxclip  = maxclip  || 10;
	this.maxres   = maxres   || this.maxclip * 10;

	this.clipammo = this.maxclip;
	this.resammo  = this.maxres;

//	this.gun = new Entity();	
//	this.hand = new Entity();	
}

Weapon.prototype.clone = function()
{
	var w = new Weapon(this.tag, this.maxclip, this.maxres);
	v.clipammo = this.clipammo;
	v.resammo  = this.resammo;
	return w;
}

File: server/Weapon.py
#
# Weapon.py
#

class Weapon:
	def __init__(self, name, damage = 10, firerate = 10, maxclip = 10, maxres = 100):
		self.name     = name
		self.damage   = damage
		self.firerate = firerate
		self.maxclip  = maxclip
		self.maxres   = maxres
		
		self.clipammo = maxclip
		self.resammo  = maxres
		
	def Reload(self):
		if self.clipammo != self.maxclip and self.resammo > 0:
			total = self.clipammo + self.resammo
			newclip = total < self.maxclip if total else self.maxclip

			self.resammo -= newclip - self.clipammo
			self.clipammo = newclip

			return True
		else:
			return False

	def Fire(self):
		if self.clipammo > 0:
			self.clipammo -= 1
			return True
		else:
			return False
